arch/s390/s390_functions.h

```diff
@@ -5,10 +5,22 @@
 #ifndef S390_FUNCTIONS_H_
 #define S390_FUNCTIONS_H_
 
+#define XSTR(s) STR(s)
+#define STR(s) #s
+
 #ifdef S390_CRC32_VX
 uint32_t crc32_s390_vx(uint32_t crc, const uint8_t *buf, size_t len);
+
+#ifdef __clang__
+#  if ( (__clang_major__ >=18) && (__clang_major__ <= 19 && __clang_minor__ <= 1 && __clang_patchlevel__ < 2) )
+#    pragma message "CRC32-VX extension is broken on current Clang version (" XSTR(__clang_major__) "." XSTR(__clang_minor__) "." XSTR(__clang_patchlevel__) "). " \
```

Comment:
I wouldn't bother trying to print the current version. It would remove the need for the macros.